rasa/cli/interactive.py

```diff
@@ -68,57 +83,30 @@ def _set_not_required_args(args: argparse.Namespace) -> None:
     args.store_uncompressed = False
 
 
-def interactive_core(args: argparse.Namespace):
-    _set_not_required_args(args)
-
-    if args.model is None:
-        zipped_model = train.train_core(args)
-    else:
-        zipped_model = get_provided_model(args.model)
-
-    perform_interactive_learning(args, zipped_model)
-
-
-def perform_interactive_learning(args, zipped_model):
+def perform_interactive_learning(args: argparse.Namespace, zipped_model: Text) -> None:
     from rasa.core.train import do_interactive_learning
 
-    if zipped_model and os.path.exists(zipped_model):
-        args.model = zipped_model
+    args.model = zipped_model
 
-        with model.unpack_model(zipped_model) as model_path:
-            args.core, args.nlu = model.get_model_subdirectories(model_path)
-            stories_directory = data.get_core_directory(args.data)
-
-            args.endpoints = get_validated_path(
-                args.endpoints, "endpoints", DEFAULT_ENDPOINTS_PATH, True
+    with model.unpack_model(zipped_model) as model_path:
+        args.core, args.nlu = model.get_model_subdirectories(model_path)
+        if args.nlu is not None and args.core is None:
+            utils.print_error_and_exit(
+                "Can not run interactive learning on an NLU-only model."
             )
+        stories_directory = data.get_core_directory(args.data)
```

Comment:
please also use the fileimporter for that. Ideally, the one used above.

Reply:
this line actually doesn't make sense i think. from following it through it seems like that stories directory is used for visualization. but it comes from `args.data` and not `args.core` so the stories that are visualized may not match up with the model given -- it should be the latter right?

Reply:
I think just replacing `stories_directory` with `args.core` in L 103 should fix that?

Reply:
yup, just tried it loading a moodbot model from the formbot directory, def a bug 😅

Reply:
I will report that as a separate bug though

Reply:
I'll skip using the fileimporter here since it should come from the model anyway. Not sure how that would work actually since the stories aren't stored in the model, right? reporting that separately here https://github.com/RasaHQ/rasa/issues/4981

Reply:
I don’t think the model contains any training data anymore by default. I think you need to use data passed in separately to the model

Reply:
> yup, just tried it loading a moodbot model from the formbot directory, def a bug 😅

I'm not sure if it's a bug. I mean it shows you which stories are already covered in the training data. If you reference the wrong data then it's your fault 🤔

Reply:
Hm okay not sure i'd agree because if you don't pass data, it just auto-picks the `--data` path which could be confusing but that's a separate thing. If I used it and passed a model, i (clearly) expect it to match the model I gave, but can see how that isn't technically feasible.

maybe we should throw an INFO or something about where the visualization data is coming from if the model is passed explicitly

Reply:
Hm okay not sure i'd agree because if you don't pass data, it just auto-picks the --data path which could be confusing but that's a separate thing. If I used it and passed a model, i (clearly) expect it to match the model I gave, but can see how that isn't technically feasible.

maybe we should throw an INFO or something about where the visualization data is coming from if the model is passed explicitly

Reply:
> maybe we should throw an INFO or something about where the visualization data is coming from if the model is passed explicitly

sounds like a plan 👍